src/book/02-system/01-nubots/10-mathematics.mdx

```diff
@@ -0,0 +1,258 @@
+
+$$
+\mathbf{r}^b_{CD} = \mathbf{R}^b_a \cdot \mathbf{r}^a_{CD}
+$$
+
+This is written in the NUbots codebase as `rCDb = Rba * rCDa`. Note that the notation is $\mathbf{R}_{from}^{to}$.
+
+## Homogeneous Transformations
+
+A homogeneous transformation matrix in three dimensions is a 4x4 matrix containing a **rotation** component and a **translation** component.
+
+If we have some rotation $\mathbf{R}_a^b$ and a translation component $\mathbf{r}^b_{AB}$, then we have a transformation matrix that transforms from space $a$ to space $b$:
+
+$$
+\mathbf{H}_a^b = \begin{bmatrix}
+\mathbf{R}_a^b & \mathbf{r}_{AB}^b \\
+\mathbf{0} & 1
+\end{bmatrix}
+$$
+
+This is written in the NUbots codebase as `Hba`.
+
+The vector $\mathbf{r}_{AB}^b$ is the vector from B (the origin of space b) to A (the origin of space a) in b space, which would be written in the codebase as `rABb`.
+
+The reason for this is that if we apply the transform $H^b_a$ to some point $\rho^a$, then $\rho^a$ will first be rotated by $R_a^b$, giving $\rho^b$. $\rho^b$ will then be translated by some vector according to the transformation matrix. Since $\rho^b$ is now rotated in the $b$ coordinate system, the translation vector must also be in this $b$ coordinate system. The vector is from B to A because a point (0, 0, 0) in space $a$ should be moved to the origin of $a$ in $b$ space after transformation. Thus, the translation vector is $\mathbf{r}_{AB}^b$.
+
+This is illustrated in the following diagram.
+
+![Two three dimensional coordinate spaces, one labelled as space 'a' and the other as space 'b'. There is a vector from space 'a' to some point 'c', and is labelled 'rCAb'. There is another vector from 'b' to 'a' and is labelled 'rABb'. There is a final vector from 'b' to 'c' which is labelled 'rCBb = rABb + rCAb'](./images/maths-translation.svg)
+
+This assumes the original vector, `rCAa` **has already been rotated** into space $b$, ie we have `rCAb`.
+
+Transformation matrices **left multiply** with vectors to get a vector in the new space. This vector is **rotated and then translated**.
+
+$$
+\mathbf{r}^b_{CB} = \mathbf{H}^b_a \cdot \mathbf{r}^a_{CA}
+$$
+
+This is written in the codebase as `rCBb = Hba * rCAa`.
+
+This is equivalent to
+
+$$
+r_{CB}^b = r_{AB}^b + (R^b_a * r_{CA}^a)
+$$
+
+or in code,
+
+```cpp
+Eigen::Vector3d rCBb = rABb + (Rba * rCAa);
+```
+
+<Alert>
+
+Vectors should only be multiplied with a transformation matrix if the **last two
+letters** of the vector match. That is, the position they are going from should match the
+origin of the space they are measured in. This means that `Hab * rCBb` is
+valid, but `Hab * rCDb` is invalid. Note that the first vector is rC**B**b and the second is rC**D**b.
+
+</Alert>
+
+If you would like the vector **only rotated**, extract the **rotational component** of the transform and apply it to the vector. This does not hold the same restriction as above, however the **spaces still need to match**.
```

Comment:
Does not hold the same restriction as above is a little confusing. I would just say that only the space needs to match.